src/cli/repl.rs

```diff
@@ -207,16 +213,17 @@ impl Repl<F> {
                     let env_out = self.store.get_z_expr(&output.env, &mut zstore)?.0;
                     let cont_out = self.store.get_z_cont(&output.cont, &mut zstore)?.0;
 
-                    let proof_claim = Self::proof_claim(
+                    let claim = Self::proof_claim(
                         &mut self.store,
                         (input.expr, output.expr),
                         (input.env, output.env),
                         (cont.parts(), cont_out.parts()),
                     );
-                    let commitment = Commitment::new(F::ZERO, proof_claim, &mut self.store)?;
-                    let proof_id = &commitment.hidden.value().hex_digits();
 
-                    let proof_path = proof_path(proof_id);
+                    let claim_comm = Commitment::new(F::ZERO, claim, &mut self.store)?;
```

Comment:
It would be better not to put `F::ZERO` here, if the point is to use a default, implicitly non-hiding commitment such as `(commit …)` provides. Instead, I would add constructors for `Commitment` that mirror the Lurk functions they mimic and are documented appropriately. In other words, it's 'okay' to use `ZERO` as an application-specific default, but if that's the intention, it's much better to use the language-provided construct for this. Although we won't do so, the default secret *could* be changed, but the value of one existing is partially in that everyone will agree on it. So let's try to move away from burying magic constants (even good, obvious ones like `ZERO`) throughout application code. Especially since values like `F::ZERO` get used for many different purposes, we risk missing the actual logic for its use in some particular place and therefore failing to notice a misuse.

Reply:
Looking at what you did: my point was not to formalize a 'default secret', but rather for you (as an application-level Lurk user here — which is effectively how the REPL/CLI are acting) to use the existing vocabulary. We already have two verbs, `hide` and `commit`. The latter (`commit`) does what you want here.

You already call `store.hide(…)`. My suggestion would be to add a new constructor that doesn't take a secret and instead calls `store.commit(…)`. The point is that nothing about how hide/commit work should be exposed to or manipulated at this level. This is *why* we provide them as language primitives.

In other words, if you want a 'non-hiding commitment', *please* use the Lurk API for creating one. That is via the `(commit …)` function, accessible from Rust via `Store::commit()`. I'd probably just rename your `new` constructor to `hide` and call the new one `commit` for reasons I hope are obvious.